Content.Server/Wieldable/WieldableSystem.cs

```diff
@@ -0,0 +1,280 @@
+﻿using Content.Server.DoAfter;
+using Content.Server.Hands.Components;
+using Content.Server.Hands.Systems;
+using Content.Server.Items;
+using Content.Server.Weapon.Melee;
+using Content.Server.Wieldable.Components;
+using Content.Shared.Hands;
+using Content.Shared.Hands.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Notification.Managers;
+using Content.Shared.Throwing;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+
+namespace Content.Server.Wieldable
+{
+    public class WieldableSystem : EntitySystem
+    {
+        [Dependency] private readonly DoAfterSystem _doAfter = default!;
+        [Dependency] private readonly HandVirtualItemSystem _virtualItemSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<WieldableComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<WieldableComponent, ItemWieldedEvent>(OnItemWielded);
+            SubscribeLocalEvent<WieldableComponent, ItemUnwieldedEvent>(OnItemUnwielded);
+            SubscribeLocalEvent<WieldableComponent, RemovedFromHandEvent>(OnItemLeaveHand);
+            SubscribeLocalEvent<WieldableComponent, VirtualItemDeletedEvent>(OnVirtualItemDeleted);
+
+            SubscribeLocalEvent<IncreaseDamageOnWieldComponent, MeleeHitEvent>(OnMeleeHit);
+        }
+
+        private void OnUseInHand(EntityUid uid, WieldableComponent component, UseInHandEvent args)
+        {
+            if (args.Handled)
+                return;
+            if(!component.Wielded)
+                AttemptWield(uid, component, args.User);
+            else
+                AttemptUnwield(uid, component, args.User);
+        }
+
+        public bool CanWield(EntityUid uid, WieldableComponent component, IEntity user)
```

Comment:
Maybe consider giving this an optional `quiet=false` argument, so that other systems or verbs can check if a user `CanWield()` without generating visible popup messages or needing to duplicate code?

Reply:
i feel like i remember adding this but i guess i didnt so i'll do it now